sklearn/neighbors/tests/test_neighbors.py

```diff
@@ -1058,9 +1077,17 @@ def test_valid_brute_metric_for_auto_algorithm():
     require_params = ['mahalanobis', 'wminkowski', 'seuclidean']
     for metric in VALID_METRICS['brute']:
         if metric != 'precomputed' and metric not in require_params:
-            nn = neighbors.NearestNeighbors(n_neighbors=3, algorithm='auto',
-                                            metric=metric).fit(X)
-            nn.kneighbors(X)
+            if metric != 'haversine':
+                nn = neighbors.NearestNeighbors(n_neighbors=3,
+                                                algorithm='auto',
+                                                metric=metric).fit(X)
```

Comment:
Nit: maybe factorize the estimator definition outside of the `if`?

Reply:
A very good nit.